src/models/regression.py

```diff
@@ -131,18 +132,25 @@ def explain(self, data: TrainData) -> Tuple[np.ndarray, ...]:
             self.explainer: shap.LinearExplainer = shap.LinearExplainer(
                 self.model, (mean, None), feature_dependence='independent')
 
-        x = data.historical
-        batch, timesteps, dims = x.shape[0], x.shape[1], x.shape[2]
-        reshaped_x = self._concatenate_data(data)
+        if x is None:
+            test_arrays_loader = DataLoader(data_path=self.data_path, batch_file_size=1,
+                                            experiment=self.experiment,
+                                            shuffle_data=False, mode='test')
+            _, val = list(next(iter(test_arrays_loader)).items())[0]
+            x = val.x
 
+        reshaped_x = self._concatenate_data(x)
         explanations = self.explainer.shap_values(reshaped_x)
 
-        historical = explanations[:, :timesteps * dims]
-        additional = explanations[:, timesteps * dims:]
+        if save_shap_values:
+            analysis_folder = self.model_dir / 'analysis'
+            if not analysis_folder.exists():
+                analysis_folder.mkdir()
 
-        # TODO: properly split the additional arrays
+            np.save(analysis_folder / f'shap_values.npy', explanations)
```

Comment:
How do we connect which features correspond to which indices in the `.npy` arrays?

Reply:
aah... still a WIP!